src/crypto/qat/qcccrypto.cc

```diff
@@ -260,57 +313,43 @@ bool QccCrypto::destroy() {
   return true;
 }
 
-void QccCrypto::do_crypt(qcc_thread_args *thread_args) {
-  auto entry = thread_args->entry;
-  qcc_op_mem[entry].op_result = cpaCySymPerformOp(qcc_inst->cy_inst_handles[entry],
-      NULL,
-      qcc_op_mem[entry].sym_op_data,
-      qcc_op_mem[entry].src_buff_list,
-      qcc_op_mem[entry].src_buff_list,
-      NULL);
-  qcc_op_mem[entry].op_complete = true;
-  free(thread_args);
-}
-
-bool QccCrypto::perform_op(unsigned char* out, const unsigned char* in,
-    size_t size, uint8_t *iv, uint8_t *key, CpaCySymCipherDirection op_type)
+bool QccCrypto::perform_op_batch(unsigned char* out, const unsigned char* in, size_t size,
+    Cpa8U *iv,
+    Cpa8U *key,
+    CpaCySymCipherDirection op_type,
+    optional_yield y)
 {
   if (!init_called) {
     dout(10) << "QAT not intialized yet. Initializing now..." << dendl;
-    if(!QccCrypto::init()) {
+    if (!QccCrypto::init(chunk_size, max_requests)) {
```

Comment:
doesn't the `QccCryptoAccel` constructor guarantee that `init()` was already called?

Reply:
> doesn't the `QccCryptoAccel` constructor guarantee that `init()` was already called?

Yes, It can be guaranteed to call. But in `init()`, the initialization of qat and the initialization of user mode memory driver may fail. So, we need to ensure that qat and memory driver have been initialized normally before executing `perform_op_batch`.